bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/QbusBridgeHandler.java

```diff
@@ -0,0 +1,295 @@
+        int port = getPort();
+
+        if (addr != null) {
+            createCommunicationObject(addr, port);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR,
+                    "Cannot connect to QbusServer with hostname " + addr);
+        }
+        int refreshInterval = getRefresh();
+        this.setupRefreshTimer(refreshInterval);
+    }
+
+    /**
+     * Sets the Bridge call back
+     */
+    private void setBridgeCallBack() {
+        if (qbusComm != null) {
+            qbusComm.setBridgeCallBack(this);
+        }
+    }
+
+    /**
+     * Create communication object to Qbus server and start communication.
+     *
+     * @param addr : IP address of Qbus server
+     * @param port : Communication port of QbusServer
+     */
+    private void createCommunicationObject(InetAddress addr, int port) {
+        scheduler.submit(() -> {
+
+            qbusComm = new QbusCommunication();
+
+            setBridgeCallBack();
+
+            if (qbusComm != null) {
+                qbusComm.startCommunication();
+            }
+
+            if (qbusComm != null) {
+                if (!qbusComm.communicationActive()) {
+                    qbusComm = null;
+                    bridgeOffline("No communication with Qbus Server");
+                    return;
+                }
+            }
+
+            if (qbusComm != null) {
+                if (!qbusComm.clientConnected()) {
+
+                    qbusComm = null;
+                    bridgeOffline("No communication with Qbus Client");
+                    return;
+                }
+            }
+
+            updateStatus(ThingStatus.ONLINE);
+
+        });
+    }
+
+    /**
+     * Take bridge offline when error in communication with Qbus server. This method can also be
+     * called directly from {@link QbusCommunication} object.
+     */
+    public void bridgeOffline(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, message);
+    }
+
+    /**
+     * Put bridge online when error in communication resolved.
+     */
+    public void bridgeOnline() {
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    /**
+     * Schedule future communication refresh.
+     *
+     * @param interval_config Time before refresh in minutes.
+     */
+
+    private void setupRefreshTimer(int refreshInterval) {
+        ScheduledFuture<?> timer = refreshTimer;
+        if (timer != null) {
+            timer.cancel(true);
+            refreshTimer = null;
+        }
+
+        if (refreshInterval == 0) {
+            return;
+        }
+
+        // This timer will check connection with server and client periodically
+        logger.info("Check communication with Server and Client every {} min", refreshInterval);
+        refreshTimer = scheduler.scheduleWithFixedDelay(() -> {
+            logger.info("Checking connection with Qbus Server & Client.");
```

Comment:
Please remove these kind of log messages. I strongly encourage you to use the debugger to check where your program runs. This saves a lot of headache.